citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -379,4 +384,14 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
         }
         return result;
     }
-}
+
+    @Override
+    public void deleteCitations(List<String> citationList) throws CannotFindCitationException {
+        for(String citation : citationList) {
+            Optional<ICitation> oldCitation = citationRepository.findByKey(citation);
+            if(oldCitation.isPresent()) {
+                citationRepository.delete((Citation)oldCitation.get());
```

Comment:
use the citationstore as the indexer needs to be updated when deleting citations

Reply:
also, you need to delete the citations from zotero, don't you?

Reply:
and when you do, make sure to make one request to zotero with all citations to be deleted not multiple (i'm fairly certain that's possible).